src/analysis/processing/qgsalgorithmfillnodata.cpp

```diff
@@ -0,0 +1,158 @@
+/***************************************************************************
+                         qgsalgorithmfillnodata.cpp
+                         ---------------------
+    begin                : January 2020
+    copyright            : (C) 2020 by Clemens Raffler
+    email                : clemens dot raffler at gmail dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsalgorithmfillnodata.h"
+#include "qgsrasterfilewriter.h"
+
+///@cond PRIVATE
+
+QString QgsFillNoDataAlgorithm::name() const
+{
+  return QStringLiteral( "fillnodata" );
+}
+
+QString QgsFillNoDataAlgorithm::displayName() const
+{
+  return QObject::tr( "Fill NoData cells" );
+}
+
+QStringList QgsFillNoDataAlgorithm::tags() const
+{
+  return QObject::tr( "noData,NoData,data,cells,fill,set" ).split( ',' );
+}
+
+QString QgsFillNoDataAlgorithm::group() const
+{
+  return QObject::tr( "Raster tools" );
+}
+
+QString QgsFillNoDataAlgorithm::groupId() const
+{
+  return QStringLiteral( "rastertools" );
+}
+
+void QgsFillNoDataAlgorithm::initAlgorithm( const QVariantMap & )
+{
+  addParameter( new QgsProcessingParameterRasterLayer( QStringLiteral( "INPUT" ), QStringLiteral( "Raster input" ) ) );
+  addParameter( new QgsProcessingParameterBand( QStringLiteral( "BAND" ), QObject::tr( "Band Number" ), 1, QStringLiteral( "INPUT" ) ) );
+  addParameter( new QgsProcessingParameterNumber( QStringLiteral( "FILL_VALUE" ), QObject::tr( "Fill value" ), QgsProcessingParameterNumber::Double, 1, false, 1, 10000000 ) );
```

Comment:
```suggestion
  addParameter( new QgsProcessingParameterNumber( QStringLiteral( "FILL_VALUE" ), QObject::tr( "Fill value" ), QgsProcessingParameterNumber::Double, 1, false ) );
```

I don't think we should have a fixed min/max here -- we need to allow negative numbers and potentially there's use cases where 0 or > 10000000 would be required.